src/garage/sampler/rl2_sampler.py

```diff
@@ -0,0 +1,195 @@
+"""BatchSampler which uses VecEnvExecutor to run multiple environments."""
+from collections import OrderedDict
+import itertools
+import pickle
+import time
+
+from dowel import logger, tabular
+import numpy as np
+
+from garage.experiment import deterministic
+from garage.misc import tensor_utils
+from garage.misc.prog_bar_counter import ProgBarCounter
+from garage.sampler.batch_sampler import BatchSampler
+from garage.sampler.stateful_pool import singleton_pool
+from garage.sampler.utils import truncate_paths
+from garage.sampler.vec_env_executor import VecEnvExecutor
+
+
+class RL2Sampler(BatchSampler):
+    """BatchSampler which uses VecEnvExecutor to run multiple environments.
+
+    This sampler is for RL^2. See https://arxiv.org/pdf/1611.02779.pdf.
+
+    Args:
+        algo (garage.np.algos.RLAlgorithm): An algorithm instance.
+        env (garage.envs.GarageEnv): An environement instance.
+        meta_batch_size (int): Number of meta batch for sampling.
+        episode_per_task (int): Number of episode for each task.
+        n_envs (int): Number of environment instances to setup.
+            This parameter has effect on sampling performance.
+
+    """
+
+    def __init__(self,
+                 algo,
+                 env,
```

Comment:
envs? shouldn't this be multiple environments?

Reply:
Are we only using one environment and set it with different tasks, or different environments?

Reply:
different environments. see metaworld ML10 and ML45.

Reply:
I see that both ProMP and original MAML implementation has a single-task interface. Can this functionality - train on different environments - supported through an env wrapper?

Reply:
Multiple environments is the normal case, single-task environments is the special case. You can use the wrapper for now to get your code merged, but @krzentner designed the sampler to specifically support this.